### PR TITLE
feat: primitive TypeScript support

### PR DESCRIPTION
**Pull request by buschtoens**
This PR enables primitive TypeScript support by:

- processing `.js` _and_ `.ts` files
- adding [`esdoc-typescript-plugin`](https://github.com/esdoc/esdoc-plugins/tree/master/esdoc-typescript-plugin)

This solution is not perfect, but _good enough_ to at least enable documenting TypeScript code, until [`ember-cli-addon-docs-typedoc`](https://github.com/typed-ember/ember-cli-addon-docs-typedoc) is or `ember-cli-addon-docs-tsdoc` are a real thing.

### CODE REVIEW

**Comment by buschtoens** [comment]
It seems that for TypeScript files all `@tags` are ignored.

For instance, this yields:

```ts
class Foo {
  /**
   * this is method3.
   * @param n - this is the param desc.
   * @return this is the return desc.
   * @private
   */
  private method3(n: number): string {}
}
```


image

**Comment by buschtoens** [comment]
Aha! https://github.com/esdoc/esdoc-plugins/issues/25

```ts
class Foo {
  /**
   * @desc this is method3.
   * @param n - this is the param desc.
   * @return this is the return desc.
   * @private
   */
  private method3(n: number): string {}
}
```

image

However, this the breaks the type information.

(The `@private` tag is also not automatically inferred.)

**Comment by buschtoens** [comment]
There also seems to be trouble with `Generics<T>` and `<Type>casts`: https://github.com/esdoc/esdoc-plugins/issues/74

**Comment by tstormk** [comment]
Any updates on this? We really need Typescript support for our documentation and it would be great if this addon could support it. Currently there seems to be no work around either.

It would also be great if this addon could support custom ESDoc plugins.

**Comment by rwwagner90** [comment]
@buschtoens @tstormk do either of you know if enabling the TS plugin effects normal JS output from ESDoc at all? This looks fine to me, but I would like to double check we are not messing anything up for JS users.

File: lib/preprocessors/generate-esdoc.js [hunk 1]
```diff
@@ -25,7 +26,8 @@ const ESDOC_CONFIG = {
         dynamicImport: true
       }
     },
-    { name: 'esdoc-accessor-plugin' }
+    { name: 'esdoc-accessor-plugin' },
+    { name: 'esdoc-typescript-plugin', option: { enable: true } }
```
Comment (nightire) [inline lib/preprocessors/generate-esdoc.js:30]:
```suggestion
    { name: 'esdoc-typescript-plugin' }
```

this `enable` option is `true` by default.

**Comment by nightire** [comment]
Can we cut a release with this improvement? I tested in a fairly large project which migrates some `.js` to `.ts`, and this PR works pretty well.

**Comment by rwwagner90** [comment]
@nightire I mainly wanted to make sure it did not break plain JS. Does it work well for both vanilla JS and TS for you?

**Comment by nightire** [comment]
@rwwagner90 Yes, it works well for both vanilla JavaScript and TypeScript.

**Pull request merged**